doc/source/reference/security.rst

```diff
@@ -0,0 +1,62 @@
+NumPy security considertions and reporting
+==========================================
+
+Security issues can be reported privately as described on the issue tracker.
+The [Python security reporting guidlines](https://www.python.org/dev/security/)
+are a good resource and its notes apply also to NumPy.
+
+NumPy's maintainers are not security experts.  However, we are conscientious
+about security and experts of both the code NumPy and its typical use.
+Please do notify us before creating security advisories against NumPy as
+we are happy to prioritize issues or help with assessing the severity of a bug.
+A security advisory we are not aware of beforehand can lead to a lot of work
+for all involved parties.
+
+
+Advise for using NumPy on untrusted data
+----------------------------------------
+
+A user who can freely execute NumPy (or Python) functions must be considered
+to have the same privilege as the process/Python interpreter.
+If an untrusted user has direct access to parts of the NumPy API,
+a the whole process must generally be isolated.
+
+That said, NumPy should be generally safe to use on *data* provided by
+unprivileged users and read through safe API functions (e.g. loaded from a
+text file or ``.npy`` file without pickle support).
+Malicious *values* or *data sizes* should never lead to privilege escalation. 
+
+The following points may be useful or should be noted when working with
+untrusted data:
+
+* Exhausting memory can result in an out-of-memory kill, which is a possible
+  denial of service attack.  Possible causes could be:
+
+  * Functions reading text files, which may require much more memory than
+    the original input file size.
+  * If users can create arbitrarily shaped arrays, NumPy's broadcasting means
+    that intermediate or result arrays can be much larger than the inputs.
+
+* NumPy structured dtypes allow for a large amount of complexity.  Fortunately,
+  most code fails gracefully when a structured dtype is provided unexpectedly.
+  However, code should either not allow untrusted users to provide these
+  (e.g. via ``.npy`` files) or carefully check the fields including for
+  nested structured/subarray dtypes.
+
+* Passing should generally be considered unsafe (except for data being read).
```

Comment:
Passing what?

Reply:
Oops, an earlier version had "Passing on user input", put that back.  (and adopted the below shortening as well).